src/Http/WebUtilities/src/FileBufferingWriteStream.cs

```diff
@@ -227,10 +251,10 @@ private void EnsureFileStream()
                 FileStream = new FileStream(
                     tempFileName,
                     FileMode.Create,
-                    FileAccess.ReadWrite,
-                    FileShare.Delete,
+                    FileAccess.Write,
+                    FileShare.Delete | FileShare.ReadWrite,
```

Comment:
Could use just `FileShare.Read` here to avoid the possibility of something modifying or deleting the file before the read stream is opened.

Reply:
I don't think that's a real concern but if that works sure.